packages/hyperproxy-node/src/index.js

```diff
@@ -16,35 +19,48 @@ export default class HyperproxyNode {
     async readFile(fileName) {
         await this.datResolve;
         return new Promise((resolve, reject) => {
-            this.dat.archive.readFile(fileName, (err, content) => {
-                if (err) {
-                    reject(err);
-                    return;
-                }
-                resolve(content);
-            });
+            this
+                .dat
+                .archive
+                .readFile(fileName, (err, content) => {
+                    if (err) {
+                        reject(err);
+                        return;
+                    }
+                    resolve(content);
+                });
         });
     }
 
     async _connectToHub(key) {
-        const client = new HyperproxyHubClient(WebSocket, Wrtc)
-        client.hub.subscribe(key).on('data', async ({sender, message, type}) => {
-            try {
-                if (type === 'response') {
-                    return;
-                }
-                const file = await this.readFile(message);
-                // TODO @lgvichy https://github.com/goonism/hyperproxy/issues/24
-                client.hub.broadcast(key, {
-                    sender: client.swarm.me,
-                    type: 'response',
-                    message: file
-                });
-            } catch(err) {
-                console.error("The hub experienced an error! ", err);
+        const client = new HyperproxyHubClient(WebSocket, Wrtc);
+        client
+            .hub
+            .subscribe(key)
+            .on('data', (data) => this._handleData(client, key, data));
+    }
+
+    async _handleData(client, key, {from, body, type}) {
+        try {
```

Comment:
I think maybe it would be better if we don’t have a try here but rather we just let the error be thrown and have the caller of this function handle it. What do you think?

Reply:
Since it's a callback, catching error might not be possible :-? ... Also I'd go for try/catch on the function level